modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -586,8 +590,175 @@ private void doColumnAutoSize(TableColumnBase<?,?> column, int cellsToMeasure) {
 
         // if the prefWidth has been set, we do _not_ autosize columns
         if (prefWidth == DEFAULT_COLUMN_WIDTH) {
-            TableSkinUtils.resizeColumnToFitContent(getTableSkin(), column, cellsToMeasure);
-//            getTableViewSkin().resizeColumnToFitContent(column, cellsToMeasure);
+            resizeColumnToFitContent(getTableSkin(), column, cellsToMeasure);
+        }
+    }
+
+    /**
+     * Resizes the given column based on the preferred width of all items contained in it. This can be potentially very
+     * expensive if the number of rows is large.
+     *
+     * @param tableSkin the {@code TableViewSkinBase} used to determine if we are dealing with a {@link TableView} or a
+     *                  {@link TreeTableView}
+     * @param tc        the column to resize
+     * @param maxRows   the number of rows considered when resizing. If -1 is given, all rows are considered.
+     * @since 12
+     */
+    protected void resizeColumnToFitContent(TableViewSkinBase<?,?,?,?,?> tableSkin, TableColumnBase<?, ?> tc, int maxRows) {
+        if (!tc.isResizable()) return;
+
+        Object control = tableSkin.getSkinnable();
+        if (control instanceof TableView) {
+            resizeColumnToFitContent((TableView)control, (TableColumn)tc, tableSkin, maxRows);
+        } else if (control instanceof TreeTableView) {
+            resizeColumnToFitContent((TreeTableView)control, (TreeTableColumn)tc, tableSkin, maxRows);
+        }
```

Comment:
This assumes that the only two specializations of a table control that uses `TableViewSkinBase` are `TableView` and `TreeTableView`. This is likely fine, since I doubt anyone could actually implement a new type of table whose Skin subclassed `TableViewSkinBase` without subclassing one of `TableView` or `TreeTableView` (and thus also subclassing TableViewSkin or TreeTableViewSkin).